iina/MainWindowController.swift

```diff
@@ -1114,21 +1114,33 @@ class MainWindowController: PlayerWindowController {
         }
       }
     case .windowSize:
-      if fsState.isFullscreen { return }
+      if fsState.isFullscreen {
+        if recognizer.magnification < 0 {
+          self.toggleWindowFullScreen()
+          return
+        }
+      }
 
       // adjust window size
       if recognizer.state == .began {
         // began
         lastMagnification = recognizer.magnification
+        if window.frame.height >= screenFrame.height && lastMagnification > 0 {
+          self.toggleWindowFullScreen()
+          return
+        }
       } else if recognizer.state == .changed {
         // changed
-        let offset = recognizer.magnification - lastMagnification + 1.0;
+        let offset = recognizer.magnification - lastMagnification + 1.0
         let newWidth = window.frame.width * offset
         let newHeight = newWidth / window.aspectRatio.aspect
 
         //Check against max & min threshold
         if newHeight < screenFrame.height && newHeight > minSize.height && newWidth > minSize.width {
```

Comment:
This looks like a bug in the original code, but I think you want to change all the `<` and `>` in this line to `<=` and `>=`. This is probably why you had problems getting the window to reach 100% height.